skimage/transform/steerable.py

```diff
@@ -0,0 +1,339 @@
+        """
+
+        if (len(coeff) == 1):
+            return np.fft.fftshift(np.fft.fft2(coeff[0]))
+
+        else:
+
+            Xrcos = Xrcos - 1
+
+            # ========================== Orientation residue===================
+            himask = point_op(log_rad, Yrcos, Xrcos)
+
+            lutsize = 1024
+            Xcosn = np.pi * \
+                np.array(range(-(2 * lutsize + 1), (lutsize + 2))) / lutsize
+            order = self.nbands - 1
+            const = np.power(2, 2 * order) * \
+                np.square(np.math.factorial(order)) / \
+                (self.nbands * np.math.factorial(2 * order))
+
+            Ycosn = np.sqrt(const) * np.power(np.cos(Xcosn), order)
+
+            orientdft = np.zeros(coeff[0][0].shape)
+
+            for b in range(self.nbands):
+                anglemask = point_op(
+                    angle, Ycosn, Xcosn + np.pi * b / self.nbands)
+                banddft = np.fft.fftshift(np.fft.fft2(coeff[0][b]))
+                orientdft = orientdft + \
+                    np.power(np.complex(0, 1), order) * \
+                    banddft * anglemask * himask
+
+            # ============== Lowpass component are upsampled and convoluted ===
+            dims = np.array(coeff[0][0].shape)
+
+            lostart = (np.ceil((dims + 0.5) / 2) -
+                       np.ceil((np.ceil((dims - 0.5) / 2) + 0.5) / 2))\
+                .astype(np.int32)
+
+            loend = lostart + np.ceil((dims - 0.5) / 2).astype(np.int32)
+
+            nlog_rad = log_rad[lostart[0]:loend[0], lostart[1]:loend[1]]
+            nangle = angle[lostart[0]:loend[0], lostart[1]:loend[1]]
+            YIrcos = np.sqrt(np.abs(1 - Yrcos * Yrcos))
+            lomask = point_op(nlog_rad, YIrcos, Xrcos)
+
+            nresdft = self.recon_pyr_level(
+                coeff[1:], nlog_rad, Xrcos, Yrcos, nangle)
+
+            res = np.fft.fftshift(np.fft.fft2(nresdft))
+
+            resdft = np.zeros(dims, 'complex')
+            resdft[lostart[0]:loend[0], lostart[1]:loend[1]] = nresdft * lomask
+
+            return resdft + orientdft
+
+    def recon_scf_pyramid(self, coeff):
+        """
+        Parameters
+        ----------
+        coeff : subbands of Steerable decomposition,
+            stored as a list of 'height' sublists,
+            Sublists correspond to decreasing radius level in Steerable pyramid
+            The first sublist contains the high pass subband.
+            The last sublist contains the low pass subband.
+            Intermediate sublists contains subbands from different orientations
+        """
+        if (self.height != len(coeff)):
+            raise ValueError("Height of coeff should be %d" % self.height)
+
+        for i in range(1, self.height - 1):
+            if (self.nbands != len(coeff[1])):
+                raise ValueError(
+                    "Size of intermediate sublist should be %d" % self.nbands)
+
+        M, N = coeff[0].shape
```

Comment:
`r, c` or `nr, nc` to indicate rows/columns rather than `M, N` for better consistency with skim age conventions